### ISSUE TITLE
[Bug]: AC0032 with Return Value and indented data items in reports, queries

### ISSUE THREAD

**Issue by stephanweidmann** [ALCops/Analyzers#299]
### Analyzer

ApplicationCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

AA0032

### Description

I have a local procedure with a record as a return value. AC0031 also recognize the read, so tabledata MySetup = r was added.

But after that, AC0032 does not recognize the read and marks the permission as "Permission tabledata mysetup is declared but this objects performs no database operations on table mysetup

    local procedure GetSetup() MySetup: Record "MySetup"
    begin
        if not MySetup.Get(UserId()) then
            MySetup.Get('');
    end;    

### AL code to reproduce

```al
codeunit 55129 STWDemo
{
    Permissions =
        tabledata "User Setup" = r;

    local procedure GetUserSetup() UserSetup: Record "User Setup"
    begin
        if not UserSetup.Get(UserId()) then
            UserSetup.Get('');
    end;
}
```

**Comment by stephanweidmann** [comment]
Addition: 
On queries or reports, indented dataitems (level 2 or more) are also shown with AC0032

**Comment by ralf-escher** [comment]
@stephanweidmann 
similar to my issue #295?

**Comment by stephanweidmann** [comment]
@ralf-escher  yes, might be - but in your example, i cannot see that ac0031 is triggered in the first place. a parameter variable is already read - no permissions will be needed.

**Comment by Arthurvdv** [comment]
@stephanweidmann, thanks for reporting this issue! I believe this should now be resolved in `v0.8.4` of ALCops.

Would you be able to update to this version and let me know if it fixes both issues on your side as well?

**Comment by stephanweidmann** [comment]
Hi @Arthurvdv,

I updated, and XMLPort and Reports are fine.
But I missed the xmlPorts - a nested tableelement in an xmlport still throws AC0032.

Thank you for your fast response and feedback. (should I open another ticket for the xmlports?)

**Comment by Arthurvdv** [comment]
> But I missed the xmlPorts - a nested tableelement in an xmlport still throws AC0032.

Thanks, I've missed this one and should now be fixed with the release of version `v0.8.6` of ALCops.

**Comment by stephanweidmann** [comment]
Yes, it is - thank you.
I will close this ticket.
